packages/discord.js/typings/index.d.ts

```diff
@@ -3913,14 +3913,16 @@ export class ApplicationCommandManager<
     guildId: Snowflake,
   ): Promise<ApplicationCommand>;
   public fetch(
-    id: Snowflake,
-    options: FetchApplicationCommandOptions & { guildId: Snowflake },
+    options: Snowflake | (Omit<FetchApplicationCommandOptions, 'guildId'> & { id: Snowflake }),
+  ): Promise<ApplicationCommandScope>;
+  public fetch(
+    options: FetchApplicationCommandOptions & { id: Snowflake; guildId: Snowflake },
   ): Promise<ApplicationCommand>;
-  public fetch(options: FetchApplicationCommandOptions): Promise<Collection<Snowflake, ApplicationCommandScope>>;
-  public fetch(id: Snowflake, options?: FetchApplicationCommandOptions): Promise<ApplicationCommandScope>;
   public fetch(
-    id?: Snowflake,
-    options?: FetchApplicationCommandOptions,
+    options: Omit<FetchApplicationCommandOptions, 'id'> & { guildId: Snowflake },
+  ): Promise<Collection<Snowflake, ApplicationCommandScope>>;
+  public fetch(
+    options?: Omit<FetchApplicationCommandOptions, 'id' | 'guildId'>,
   ): Promise<Collection<Snowflake, ApplicationCommandScope>>;
```

Comment:
`guildId` is optional in `FetchApplicationCommandOptions`. We can just make the `options` parameter optional to merge both these overloads.

```suggestion
    options?: Omit<FetchApplicationCommandOptions, 'id'>,
  ): Promise<Collection<Snowflake, ApplicationCommandScope>>;
```